planet/src/main/webapp/js/placesItinerary.js

```diff
@@ -13,56 +13,140 @@
 // limitations under the License.
 
 // This file contains functions related to the places part of the itinerary page
+
+import TimeRange from './TimeRange.js';
+
+// Declare global functions.
 window.openAddPlaceForm = openAddPlaceForm;
 window.closeAddPlaceForm = closeAddPlaceForm;
 
+// Declare global variables/constants.
 const database = firebase.database();
+let map = new google.maps.Map(document.getElementById('empty-map'));
+let savedPlaces;  // Object that contains all the places information
 
 export function renderPlaces() {
+    savedPlaces = {};
     const userId = firebase.auth().currentUser.uid;
     const placesRef = database.ref('users/' + userId + '/places');
     placesRef.once('value', (placesSnapshot) => {
         const placesContainer = document.getElementById('places');
         placesContainer.innerHTML = '';
-        placesSnapshot.forEach((childPlace) => {
-            let placeDetailsRef = database.ref('places/' + childPlace.val());
-            placeDetailsRef.once('value', (placeDetailsSnapshot) => {
-                let placeObject = placeDetailsSnapshot.val();
-                let placeElement = createPlaceElement (placeDetailsSnapshot.key,
-                                                placeObject.name, 
-                                                placeObject.address);
-                placesContainer.appendChild(placeElement);
-            });
-        });
+        const savedPlaces = placesSnapshot.val();
+        for (const placeId in savedPlaces) {
+            // Make request with fields
+            let placeRequest = {
+                placeId: placeId,
+                fields: ['place_id','name','rating','formatted_address',
+                    'opening_hours','photos','url']
+            };
+            let service = new google.maps.places.PlacesService(map);
+            service.getDetails(placeRequest, renderPlaceDetailsCallback); 
+        }
     });
 }
 
-function createPlaceElement(ref, name, address) {
+function renderPlaceDetailsCallback(place, status) {
+    if (status == google.maps.places.PlacesServiceStatus.OK) {
+        const placesContainer = document.getElementById('places');
+        const placeElement = createPlaceElement(place);
+        placesContainer.appendChild(placeElement);
+    } else {
+        alert( 'Too many requests at the same time! please try again in another 10s. ');
```

Comment:
If I'm understanding correctly, the user can't directly control how many requests they're making. An alert is useful if the user can take action to fix the problem, but they can't really do that in this case.

Instead, I think we could try to hide or work around the issue so that we never issue too many requests at once. We can do that in a future PR, but in the meantime I'd still recommend removing the alert to avoid confusion.

Reply:
Okay!